File: app/src/main/java/android/prada/lab/goodthingmap/model/UserMessage.java
package android.prada.lab.goodthingmap.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

/**
 * Created by prada on 6/28/14.
 */
public class UserMessage implements Parcelable {

    @SerializedName("messageId")
    private int id;
    @SerializedName("userId")
    private String userId;
    @SerializedName("message")
    private String message;
    @SerializedName("time")
    private long time;

    private UserMessage(String comment) {
        id = -1;
        userId = "";
        time = System.currentTimeMillis();
        message = comment;
    }

    public static UserMessage newInstance(String comment) {
        UserMessage instance = new UserMessage(comment);
        return instance;
    }

    public long getTime() {
        return time;
    }

    public void setTime(long time) {
        this.time = time;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    protected UserMessage(Parcel in) {
        id = in.readInt();
        userId = in.readString();
        message = in.readString();
        time = in.readLong();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(id);
        dest.writeString(userId);
        dest.writeString(message);
        dest.writeLong(time);
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<UserMessage> CREATOR = new Parcelable.Creator<UserMessage>() {
        @Override
        public UserMessage createFromParcel(Parcel in) {
            return new UserMessage(in);
        }

        @Override
        public UserMessage[] newArray(int size) {
            return new UserMessage[size];
        }
    };
}

File: app/src/main/kotlin/android/prada/lab/goodthingmap/model/UserMessage.kt
package android.prada.lab.goodthingmap.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

class UserMessage() : Parcelable {
    companion object {
        @JvmStatic
        fun newInstance(comment: String): UserMessage = UserMessage().apply {
            id = -1
            userId = ""
            time = System.currentTimeMillis()
            message = comment
        }

        @JvmField
        val CREATOR: Parcelable.Creator<UserMessage> = object : Parcelable.Creator<UserMessage> {
            override fun createFromParcel(source: Parcel): UserMessage = UserMessage(source)
            override fun newArray(size: Int): Array<UserMessage?> = arrayOfNulls(size)
        }
    }

    @field:SerializedName("messageId")
    var id: Int = 0
    @field:SerializedName("userId")
    var userId: String? = null
    @field:SerializedName("message")
    var message: String? = null
    @field:SerializedName("time")
    var time: Long = 0L

    private constructor(source: Parcel) : this() {
        id = source.readInt()
        userId = source.readString()
        message = source.readString()
        time = source.readLong()
    }

    override fun describeContents(): Int = 0

    override fun writeToParcel(destination: Parcel, flags: Int) {
        destination.writeInt(id)
        destination.writeString(userId)
        destination.writeString(message)
        destination.writeLong(time)
    }
}
